File: Stack/Implementation/LinkedList/Stack.java
package Stack.Implementation.LinkedList;

public class Stack {

    StackNode root;

    static class StackNode {
        int data;
        StackNode next;

        StackNode(int data) {
            this.data = data;
        }

    }

    boolean isEmpty() {
        return (root == null) ? true : false;
    }

    void push(int data) {

        StackNode newNode = new StackNode(data);

        if (root == null) {
            root = newNode;
        } else {
            StackNode temp = root;
            root = newNode;
            newNode.next = temp;
        }

        System.out.println(data + " : pushed into stack ");
    }

    int pop() {
        int popped = Integer.MIN_VALUE;
        if (root == null) {
            System.out.println("Stack is empty");
        } else {
            popped = root.data;
            root = root.next;
        }
        return popped;
    }

    int peek() {

        if (root == null) {
            System.out.println("Stack is empty");
            return Integer.MIN_VALUE;
        } else {
            return root.data;
        }
    }

    public static void main(String[] args) {
        Stack s = new Stack();

        System.out.println("Peek() : " + s.peek());

        for (int i = 0; i < 10; i++)
            s.push(i * 100);

        for (int i = 0; i < 5; i++) {
            System.out.println(s.pop());
        }

        System.out.println("isEmpty() : " + s.isEmpty());

        System.out.println("Peek() : " + s.peek());

        for (int i = 0; i < 10; i++) {
            s.pop();
        }

    }

}

File: Stack/Implementation/LinkedList/Stack.cpp
#include <iostream>
using namespace std;

class StackNode
{
public:
    int data;
    StackNode *next;

    StackNode(int data)
    {
        this->data = data;
        this->next = NULL;
    }
};

class Stack
{

private:
    StackNode *root;

public:
    Stack()
    {
        root = NULL;
    }

    // To check if the Stack is Empty or not
    bool isEmpty()
    {
        return (root == NULL) ? true : false;
    }

    // To push element to the Stack
    void push(int data)
    {

        if (root == NULL)
        {
            root = new StackNode(data);
        }
        else
        {

            StackNode *newStackNode = new StackNode(data);
            newStackNode->next = root;
            root = newStackNode;
        }

        cout << data << " is Pushed to the Stack\n";
    }

    // To pop element from the Stack
    int pop()
    {
        if (isEmpty())
        {
            cout << "Stack Underflow\n";
            return INT32_MIN;
        }
        else
        {
            int popped = root->data;
            root = root->next;
            return popped;
        }
    }

    // To peek the top of the Stack
    int peek()
    {
        if (isEmpty())
        {
            cout << "Stack Underflow\n";
            return INT32_MIN;
        }
        else
        {
            return root->data;
        }
    }
};

int main()
{

    Stack *s1 = new Stack();

    cout << "IsEmpty: " << s1->isEmpty() << "\n";

    s1->push(1);
    s1->push(2);
    s1->push(3);
    s1->push(4);

    cout << "Peek: " << s1->peek() << "\n";

    s1->pop();

    cout << "Peek: " << s1->peek() << "\n";



    return 0;
}